mcx/distributions/mvnormal.py

```diff
@@ -0,0 +1,40 @@
+from jax import numpy as np
+from jax import random, scipy
+
+from mcx.distributions import constraints
+from mcx.distributions.distribution import Distribution
+from mcx.distributions.shapes import broadcast_batch_shape
+
+
+class MvNormal(Distribution):
+    params_constraints = {
+        "mu": constraints.real_vector,
+        "covariance_matrix": constraints.positive_definite,
+    }
+    support = constraints.real_vector
+
+    def __init__(self, mu, covariance_matrix):
+        self.event_shape = broadcast_batch_shape(np.shape(mu))
```

Comment:
Shapes are trickier with multivariate distributions. `event_shape` refers to the shape of one sample from the distribution; it is here different from 1. It is indeed specified by `mu`, which is necessarily a vector. But we also want to determine the `batch_shape` (number of iid samples returned) with `mu`.

So say we want to sample from 4 iid random variables with a Multivariate Normal distribution of mean vector `[1., 1.]`. Naturally we want to write something like:

`mu = [[1,1], [1,1], [1,1], [1,1]]`

of shape `(4, 2)`. We can thus choose the *convention* `event_shape = broadcast_batch_shape(np.shape(mu)[-1])`.

Reply:
That was helpful. Thanks! 
That + Eric Ma's blogpost + some mucking around and i *think* i have it.